### PR TITLE
feat(data-exploration): fix event table regressions

### LINKED ISSUE #13223: Regressions in new data table component
## Bug description

Noticed a few problems in the new feature-flagged events table in Persons (e.g. on [this marius@posthog.com page](https://app.posthog.com/person/pI40e5w84oZfhqdX4Sqf9XVmH5vVM3pyOMY1KRfe9XT#activeTab=events)):
1. Property value columns don't show anything
1. There's no "Time" column
1. There's a redundant "Person" column when on the person page
1. Similarly, there's no date separators
1. _Also_, the ClickHouse query made by the new UI takes very long (literally 20 s on the person page linked above for me, as opposed to 0.5 s with the old UI), because there no longer is a minimum timestamp filter to avoid going through too many CH parts in one query
1. The event buffer notice is missing ("Note that some events with a never-before-seen distinct ID are deliberately delayed by 60 seconds ..."
1. "More"/kebab button column is too wide, it should have `width: 0`

### `data-exploration-live-events` disabled
on

### `data-exploration-live-events` enabled
off

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Fixes https://github.com/PostHog/posthog/issues/13223

## Changes

- [x] Hides the "person" column when looking at just one person
- [x] Make the trash/more buttons `width:0`
- [x] Fixes problem with editing filters. Was an [issue in kea](https://github.com/keajs/kea/commit/97a869de5882db348cc84303b26b6e99b1320e6e) after all.
- [x] Add timestamp filter to events list query
- [ ] ? Make sure the events buffer notice is in all places
- [ ] ? Date separators
- [ ] ? Property value columns don't show anything
- [ ] ? There's no "Time" column

## How did you test this code?

Clicky click.

### CODE REVIEW

**Comment by mariusandra** [comment]
This is ready for a look. The cypress tests fail for some reason, I suspect weird caching coming from the node18 or pnpm changes.

**Review by Twixes — COMMENTED** [review]
This is _almost_ perfect now, but would be great to keep those date separators in the new component. :D
Also, it seems the event buffer notice isn't there, though I'm totally OK with doing away with that _as long as_ we only release `data-exploration-live-events` after Pipeline folks' "live events" table is live.

File: frontend/src/lib/utils.tsx [hunk 1]
```diff
@@ -1542,3 +1542,5 @@ export function downloadFile(file: File): void {
         link?.parentNode?.removeChild(link)
     }, 0)
 }
+
+export const daysAgo = (days: number): string => now().subtract(days, 'day').toISOString()
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/lib/utils.tsx:1546]: I stole it from the existing events table, and ran into some mocking trouble, so didn't reuse the new lib there. Good idea though, I just inlined this now.

File: frontend/src/queries/nodes/DataTable/renderColumnMeta.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        return { title: <PropertyKeyInfo value={key.substring(11)} type={PropertyFilterType.Event} disableIcon /> }
+    } else if (key.startsWith('context.columns.')) {
+        return { title: context?.columns?.[key.substring(16)]?.title ?? key.substring(16).replace('_', ' ') }
+    } else if (key === 'person.$delete') {
+        return { title: '', width: 0 }
+    } else if (key.startsWith('person.properties.')) {
+        // NOTE: PropertyFilterType.Event is not a mistake. PropertyKeyInfo only knows events vs elements ¯\_(ツ)_/¯
+        return { title: <PropertyKeyInfo value={key.substring(18)} type={PropertyFilterType.Event} disableIcon /> }
+    } else {
+        return { title: String(key) }
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/queries/nodes/DataTable/renderColumnMeta.tsx]: I can only assume there's a historic reason for this...

**Pull request closed without merging**